lib/Minz/ExtensionManager.php

```diff
@@ -208,10 +233,17 @@ public static function register($ext) {
 	 * The extension init() method will be called.
 	 *
 	 * @param string $ext_name is the name of a valid extension present in $ext_list.
+	 * @param 'system'|'user'|null $onlyOfType only enable if the extension matches that type. Set to null to load all.
 	 */
-	public static function enable($ext_name) {
+	private static function enable(string $ext_name, ?string $onlyOfType = null): void {
 		if (isset(self::$ext_list[$ext_name])) {
 			$ext = self::$ext_list[$ext_name];
+
+			if ($onlyOfType !== null && $ext->getType() !== $onlyOfType) {
+				// Do not enable an extension of the wrong type
+				return;
```

Comment:
In the case of extensions changing type (from `user` to `system` or vice versa), we risked enabling wrongly or two times an extension